src/aws/cis-1.3.0/rules/aws-cis-1.3.0-1.12.ts

```diff
@@ -0,0 +1,124 @@
+  
+          aws iam get-credential-report --query 'Content' --output text | base64 -d | cut -d, -f1,4,5,6,9,10,11,14,15,16
+  
+  **Ensure unused credentials do not exist:**
+  
+  2. For each user having *password_enabled* set to *TRUE*, ensure *password_last_used_date* is less than *90* days ago.
+  
+      - When *password_enabled* is set to *TRUE* and password_last_used is set to No_Information , ensure *password_last_changed* is less than *90* days ago.
+  
+  3. For each user having an *access_key_1_active* or *access_key_2_active* to *TRUE* , ensure the corresponding *access_key_n_last_used_date* is less than *90* days ago.
+  
+      - When a user having an *access_key_x_active* (where x is 1 or 2) to *TRUE* and corresponding access_key_x_last_used_date is set to *N/A*', *ensure* access_key_x_last_rotated is less than 90 days ago.`,
+  
+  rationale: 'Disabling or removing unnecessary credentials will reduce the window of opportunity for credentials associated with a compromised or abandoned account to be used.',
+  
+  remediation: `**From Console:**
+  Perform the following to manage Unused Password (IAM user console access)
+  
+  1. Login to the AWS Management Console:
+  2. Click *Services*
+  3. Click *IAM*
+  4. Click on *Users*
+  5. Click on *Security Credentials*
+  6. Select user whose *Console last sign-in* is greater than 90 days
+  7. Click *Security credentials*
+  8. In section *Sign-in credentials*, *Console password* click *Manage*
+  9. Under Console Access select *Disable*
+  10. Click *Apply*
+  
+  Perform the following to deactivate Access Keys:
+  
+  1. Login to the AWS Management Console:
+  2. Click *Services*
+  3. Click *IAM*
+  4. Click on *Users*
+  5. Click on *Security Credentials*
+  6. Select any access keys that are over 90 days old and that have been used and
+      - Click on *Make Inactive*
+  7. Select any access keys that are over 90 days old and that have not been used and
+      - Click the X to *Delete*`,
+  
+  references: [
+      'CCE-78900-8',
+      'https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#remove-credentials',
+      'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_finding-unused.html',
+      'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_passwords_admin-change-user.html',
+      'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html',
+  ],
+  gql: `{
+   queryawsIamUser {
+      id
+      arn
+      accountId
+       __typename
+      passwordLastUsed
+      accessKeyData {
+        lastUsedDate
+      }
+    }
+  }`,
+  resource: 'queryawsIamUser[*]',
+  severity: 'medium',
+  conditions: {
+    or: [
+      {
+        and: [
+          {
+            path: '@.accessKeyData',
+            isEmpty: true
+          },
+          {
+            not: {
+              path: '@.passwordLastUsed',
+              notIn: [null, 'N/A', '']
+            }
+          }
+        ]
+      },
+      {
```

Comment:
How would this evaluate if the password was last used less than 90 days ago but there was an access key that hasnt been used in more than 90 days. I think that would pass but idk if it should

Reply:
mmh yes, both conditions should be met. I'm going to add them into an "and" condition

Reply:
@tyler-dunkel  done!